File: shared/src/commonMain/kotlin/io.projectnewm.shared/models/Genre.kt
package io.projectnewm.shared.models

final enum class Genre {
    ROCK, CLASSICAL, RAP, FOLK, COUNTRY;

    companion object {
        val allCases: List<Genre> = listOf<Genre>(ROCK, CLASSICAL, RAP, FOLK, COUNTRY)
    }

    //TODO: Add localization
    val title : String
        get() = when (this) {
            FOLK -> "Folk"
            CLASSICAL -> "Classical"
            COUNTRY -> "Country"
            RAP -> "Rap"
            ROCK -> "Rock"
        }
}

File: iosApp/Modules/Helpers/Models/Genre.swift
import Foundation

public enum Genre: CaseIterable {
	case rock
	case classical
	case rap
	case folk
	case country
		
	public var title: String {
		switch self {
		case .folk: return "Folk"
		case .classical: return "Classical"
		case .country: return "Country"
		case .rap: return "Rap"
		case .rock: return "Rock"
		}
	}
}
